cmd/createredis.go

```diff
@@ -31,12 +51,22 @@ Cobra is a CLI library for Go that empowers applications.
 This application is a tool to generate the needed files
 to quickly create a Cobra application.`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("create called")
+		var tr tcloudredis.TcloudRedis
+		err := redisParseConfig(&tr)
```

Comment:
You should minimize the number of places that can access gobal variables. You should try to program interm of pure functions as much as possible, INPUT -> OUPUT, instead of INPUT + GLOBAL -> OUTPUT. e.g.
```suggestion
		redistConfig, err := redisParseConfig(redisConfName)
```
one advantage to this is that if you have to parse multiple configs, all you would have to do is call the same function multiple times

Reply:
i agree with u. 

unfortunately, how viper setup, it use init function to add the following flags:

	createRedisCmd.Flags().StringVarP(&redisConfName, "env", "e", "", "sub directory inside of ./conf what has a region's config files")

So redisConfName needs to be a global.

@riot-jayb and I want to get ride of -e after all

Reply:
I'm in favor of it